File: GFG POTD/022-September-2025/026-rotate-deque-by-k.cpp
// https://www.geeksforgeeks.org/problems/rotate-deque-by-k/1/




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N)
    // SC : O(1)
    void rotateDeque(deque<int>& dq, int type, int k) {
        while (k --) {
            if (type == 1) {
                int val = dq.back();
                dq.pop_back();
                dq.push_front(val);
            } else {
                int val = dq.front();
                dq.pop_front();
                dq.push_back(val);
            }
        }
    }
};

File: GFG POTD/022-September-2025/026-rotate-deque-by-k.java
// https://www.geeksforgeeks.org/problems/rotate-deque-by-k/1/




import java.util.*;

class Solution {
    public static void rotateDeque(Deque<Integer> dq, int type, int k) {
        while (k -- > 0) {
            if (type == 1) {
                Integer val = dq.pollLast();
                if (val != null) {
                    dq.addFirst(val);
                }
            } else {
                Integer val = dq.pollFirst();
                if (val != null) {
                    dq.addLast(val);
                }
            }
        }
    }
}